http-body-util/src/combinators/buffered.rs

```diff
@@ -0,0 +1,38 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures_util::Future;
+use http_body::Body;
+use pin_project_lite::pin_project;
+
+pin_project! {
+    /// Future that resolves into a `Buffered`.
+    pub struct Buffered<T: ?Sized> {
```

Comment:
I was at first a little confused that this had the same name as the other one. What do you think about naming them differently?

Reply:
We do somewhat the same for Frame which has the `Frame` type in the main crate. If you can come up with a better name I am game for it but BufferedFuture seemed to extra but we could do it.

Reply:
I think `BufferedFuture` is fine. At least its clear and its unlikely users will have to name this type. They're probably more interested in the output type.